packages/runtime/src/runtime/modal.tsx

```diff
@@ -188,29 +232,56 @@ export const ModalWrapper: React.FC<PropsWithChildren> = ({ children }) => {
     `
       .ant-modal-root .ant-modal-centered .ant-modal {
         top: unset;
+        max-width: 100%;
+      }
+      .ensemble-modal-${index} {
+        height: 100%;
+        max-height: 100vh;
+        max-width: 100vw;
+        ${getComponentStyles(
+          "",
+          pick(options, [
+            "height",
+            "width",
+            "position",
+            "top",
+            "left",
+            "bottom",
+            "right",
+          ]) as React.CSSProperties,
+        )}
       }
       .ensemble-modal-${index} .ant-modal-content {
-        padding: ${
-          options.padding ? options.padding : "10px 24px 24px"
-        } !important;
-        ${
-          options.backgroundColor
-            ? `background-color: ${options.backgroundColor} !important;`
-            : ""
-        } 
+        display: flex;
+        flex-direction: column;
+        ${getComponentStyles(
+          "",
+          omit(options, [
+            "width",
+            "position",
+            "top",
+            "left",
+            "bottom",
+            "right",
+          ]) as React.CSSProperties,
+        )}
         ${options.showShadow === false ? "box-shadow: none !important;" : ""}
       }
+      .ensemble-modal-${index} .ant-modal-body {
+        height: 100%;
+        overflow-y: auto;
+      }
     `;
 
   const getFullScreenStyles = (index: number): string => `
-    .ensemble-modal-${index}.ant-modal, .ensemble-modal-${index}.ant-modal .ant-modal-content {
-      height: 100vh;
-      width: 100vw;
-      margin: 0;
-      top: 0;
-    }
-    .ensemble-modal-${index} .ant-modal-body {
-      height: calc(100vh - 110px);
+    .ensemble-modal-${index}, .ensemble-modal-${index} .ant-modal-content {
+      height: 100vh !important;
+      width: 100vw !important;
+      margin: 0 !important;
+      top: 0 !important;
+      left: 0 !important;
+      bottom: 0 !important;
+      right: 0 !important;
```

Comment:
Why do you need these styles?

Reply:
I need these here when fullscreen is tapped in order to overwrite user given styles like height, top etc.

Reply:
We can streamline this right? Since you're already calculating top/left/right/bottom and applying to styles prop you don't need separate CSS rules?